### DISCUSSION TITLE
Need Help with Multi-Account One Install

### DISCUSSION THREAD

**Discussion by Abyssial** [rfvgyhn/min-ed-launcher#92] [General]
Okay so I have a copy of both steam and epic ED. I am trying to set it up so that I can use my EGS account with my Steam install while retaining the use of my Steam account. I am having troubles with retrieving the one time use epic code for launching the steam shortcut aswell as creating the shortcut with the argument that allows for this all to be possible. I have a feeling I am just not having a great day whatsoever so any help at all would be much appreciated. Thanks

**Comment by rfvgyhn** [comment]
Getting the exchange code is a pain. If you're looking to use your epic account somewhat regularly, I recommend downloading [legendary](https://github.com/derrod/legendary) and using it to generate the exchange code instead of trying to manually get it. You can then create a shortcut that launches the game through legendary and not have to worry about getting the code yourself. You might be able to do all this with [Heroic](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher) as well, but I've never used it before.

1. Download `legendary.exe` from legendary's [releases page](https://github.com/derrod/legendary/releases)
2. Place it in the same folder as MinEdLauncher.exe and EDLaunch.exe (or add it to your PATH env var)
3. Follow the [quickstart](https://github.com/derrod/legendary#quickstart) (just the auth part) to authenticate with Epic
4. Open `C:\Users\[user]\.config\legendary\installed.json` in a text editor (e.g.  notepad)
5. Overwrite the contents of the file with the following, update the path to point to the folder that contains EDLaunch.exe, and save the file
    ```
    {
      "9c203b6ed35846e8a4a9ff1e314f6593": {
        "install_path": "Z:\\EliteDangerous",
        "title": "Elite Dangerous"
      }
    }
    ```
    This tells legendary you have the game installed without having to re-download the game
7. Right click legendary.exe and create shortcut
8. Set the target to `Z:\EliteDangerous\legendary.exe launch 9c203b6ed35846e8a4a9ff1e314f6593 --skip-version-check --override-exe MinEdLauncher.exe /epic /autorun /edh4 /autoquit`
    * Update the path to match your system
    * Update the launch flags to your liking
    * `--skip-version-check` is used since you are using Steam to manage your game files
9. You can also use the _Change Icon..._ button to change the icon to an Elite icon

**Reply by Abyssial** [reply]
Thanks for the reply, I tried the steps above and when running the shortcut got the notification in the cmd of "Error: Game: 9c203b6ed35846e8a4a9ff1e314f6593 is not currently installed"

**Reply by Abyssial** [reply]
Also when looking in C:\Users\[user]\.config\legendary, installed.json was not there so i create a .json file using notepad with the code in step 5. i believe this may be the crux of the problem

**Reply by rfvgyhn** [reply]
Did any other files exist there? I believe those files should be created when legendary first runs.

**Reply by Abyssial** [reply]
So this is what is in the folder as of right now (installed.json is the file i created)

![image](https://user-images.githubusercontent.com/72520767/234730767-e8e33625-40be-45e3-837a-2ee69c7dcc94.png)

the only command i ran in legendary once it was placed in my steam ed directory was "legendary auth" no other commands were run

**Reply by rfvgyhn** [reply]
Did the `auth` step work fine or were there errors at that part too?

Instead of manually creating the installed.json file, you can also go through the normal install steps. Not sure what the differences are between the epic and steam game files, so it may not need to download much at all.

Open a terminal, change directories by running `cd Z:\EliteDangerous` (or open the folder in explorer, right click, open terminal here) and run `legendary.exe install "Elite Dangerous" --game-folder "Z:\EliteDangerous"` (using your actual path for both commands)

Worst case scenario, you can just run `legendary.exe get-token`. That will give you the exchange code and you can follow the manual process from the readme.

**Reply by Abyssial** [reply]
![image](https://user-images.githubusercontent.com/72520767/234731261-f4e0d3c0-5e9a-45e8-a95b-ce4bde5021a0.png)

**Reply by rfvgyhn** [reply]
You could also try the full json object instead of just the two properties from my example. The base_urls might not be correct for your region, but since you won't be using epic to download game updates, it shouldn't really matter.

```
{
  "9c203b6ed35846e8a4a9ff1e314f6593": {
    "app_name": "9c203b6ed35846e8a4a9ff1e314f6593",
    "base_urls": [
      "https://download.epicgames.com/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default",
      "https://download2.epicgames.com/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default",
      "https://download3.epicgames.com/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default",
      "https://download4.epicgames.com/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default",
      "https://epicgames-download1.akamaized.net/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default",
      "https://fastly-download.epicgames.com/Builds/Org/o-mlnjydnrxlmer2ducb6u6ngznq25mx/f92ad09c5eb84033825d8ee1741172b8/default"
    ],
    "can_run_offline": true,
    "egl_guid": "",
    "executable": "EDLaunch.exe",
    "install_path": "Z:\\EliteDangerous",
    "install_size": 21394317774,
    "install_tags": [],
    "is_dlc": false,
    "launch_parameters": "/Epic",
    "manifest_path": null,
    "needs_verification": false,
    "platform": "Windows",
    "prereq_info": {
      "args": "",
      "ids": [
        "DirectXJun2010Redist"
      ],
      "name": "DirectX June 2010 Redistributable",
      "path": "directx_redist/DXSETUP.exe"
    },
    "requires_ot": false,
    "save_path": null,
    "title": "Elite Dangerous",
    "version": "3.8.289583-6780"
  }
}
```

**Reply by Abyssial** [reply]
okay looks like i wrote the cmd wrong, it is now installing, appears to be downloading and installing the game and dupping files

**Reply by Abyssial** [reply]
I stopped the install becuase i feel like it will mess it up, should i validate or will that mean re-installing min-ed since steam would wipe them out?

**Reply by Abyssial** [reply]
disregard dupped files, they seem to have corrected themselves, still getting the same error after altering the installed.json

**Reply by rfvgyhn** [reply]
To get legendary to recognize the game as installed, you'll either need to try the full json in `installed.json` (making sure `install_path` is correct) or let the `legendary.exe install` process finish. Once it completes, you can verify game files within steam. Neither downloading nor verifying should require you re-do the min-ed-launcher setup. It's not a part of the game's manifest, so steam should just ignore it.

**Reply by rfvgyhn** [reply]
Another thought. In `installed.json`, are the slashes in your path escaped? `\\` instead of `\`.

**Reply by Abyssial** [reply]
no they are only single \

**Reply by Abyssial** [reply]
i tried both double \\ and single \ in the directory in installed.json, both had the error
